test/module/irohad/consensus/yac/yac_gate_test.cpp

```diff
@@ -298,6 +305,79 @@ TEST_F(YacGateTest, DifferentCommit) {
   ASSERT_TRUE(gate_wrapper.validate());
 }
 
+/**
+ * @given yac gate, in round (i, j) -> last block height is (i - 1)
+ * @when vote for round (i + 1, j) is received
+ * @then peer goes to round (i + 1, j)
+ */
+TEST_F(YacGateTest, Future) {
+  // yac consensus
+  EXPECT_CALL(*hash_gate, vote(expected_hash, _, _)).Times(1);
+
+  // generate order of peers
+  EXPECT_CALL(*peer_orderer, getOrdering(_, _))
+      .WillOnce(Return(ClusterOrdering::create({makePeer("fake_node")})));
+
+  // make hash from block
+  EXPECT_CALL(*hash_provider, makeHash(_)).WillOnce(Return(expected_hash));
+
+  block_notifier.get_subscriber().on_next(BlockCreatorEvent{
+      RoundData{expected_proposal, expected_block}, round, ledger_state});
+
+  iroha::consensus::Round future_round{round.block_round + 1,
+                                       round.reject_round};
+  PublicKey actual_pubkey("actual_pubkey");
+  auto signature = std::make_shared<MockSignature>();
```

Comment:
test/module/irohad/consensus/yac/mock_yac_crypto_provider.hpp
```suggestion
  auto signature = createSig(actual_pubkey);
```